File: 0680-valid-palindrome-ii/0680-valid-palindrome-ii.py
class Solution:
    def validPalindrome(self, s: str) -> bool:
        
        l, r = 0, len(s)-1
        while l < r:
            if s[l] != s[r]:
                s1, s2 = s[l+1: r+1], s[l:r] # skip left or right
                print(s1)
                print(s2)
                return (s1 == s1[::-1]) or (s2 == s2[::-1])
            l, r = l+1, r-1

        return True

File: 0680-valid-palindrome-ii/0680-valid-palindrome-ii.cpp
class Solution {
public:
    bool validPalindrome(string s) {
        
        	int l = 0 , r = (int)s.length()-1;
	int c = 0;
	while(l < r){
		if (s[l] == s[r])
		{
			l++;
			r--;
		}
		else{
			r --;
			c ++;
			if (c > 1)
				break;
		}
	}

	int c2= 0;
	l = 0 ,r = (int)s.length()-1;
	while(l < r){
		if (s[l] == s[r])
		{
			l++;
			r--;
		}
		else{
			l ++;
			c2 ++;
			if (c2 > 1)
				break;
		}
	}

	if (c <= 1 || c2 <= 1)
		return true;
	else
		return false;

        
        
    }
};